### PR TITLE
feat(UI): Implement new login page

### PR DESCRIPTION
**Pull request by Thebarda**
## Description

This implements the new login page.
https://user-images.githubusercontent.com/12515407/150376537-0a8eeb56-7686-4b0a-9b75-dcacb587a14a.mov

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [x] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

- Reach the centreon web page
- Fill the inputs with the alias and the password
- Click on the "Connect" button
- -> You can reach the default page
- Click on the "logout" button
- -> You're now logged out and and you are redirected to the login page 

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: tests/e2e/cypress/integration/Resources-status/common.ts [hunk 1]
```diff
@@ -1,8 +1,86 @@
+/* eslint-disable cypress/no-unnecessary-waiting */
```
Comment (bdauria) [inline tests/e2e/cypress/integration/Resources-status/common.ts:1]:
```suggestion
```

File: tests/e2e/cypress/support/index.ts [hunk 2]
```diff
@@ -1,51 +1,18 @@
...
 });
 
-after(() => {
-  return setUserTokenApiV1()
-    .then(removeResourceData)
-    .then(applyConfigurationViaClapi)
-    .then(() =>
-      cy.visit(`${Cypress.config().baseUrl}/centreon/index.php?disconnect=1`),
-    );
+Cypress.on('uncaught:exception', (err) => {
```
Comment (bdauria) [inline tests/e2e/cypress/support/index.ts:11]: This comment can be removed if unused

File: www/front_src/src/Login/index.test.tsx [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+        data: retrievedUser,
+      });
+  });
+
+  afterEach(() => {
+    mockDate.reset();
+    mockedAxios.get.mockReset();
+    mockedAxios.post.mockReset();
+    window.history.pushState({}, '', '/');
+  });
```
Comment (bdauria) [inline www/front_src/src/Login/index.test.tsx:106]:
```suggestion
  });
  
```

File: www/front_src/src/Login/index.test.tsx [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+
+    expect(screen.getByLabelText(labelAlias)).toBeInTheDocument();
+    expect(screen.getByLabelText(labelPassword)).toBeInTheDocument();
+    expect(screen.getByLabelText(labelConnect)).toBeInTheDocument();
+    expect(screen.getByLabelText(labelCentreonLogo)).toBeInTheDocument();
+    expect(screen.getByText('v. 21.10.1')).toBeInTheDocument();
+    expect(screen.getByText('Copyright © 2005 - 2020')).toBeInTheDocument();
+  });
+
+  it(`submits valid credentials when fields are filled and the "${labelConnect}" is clicked`, async () => {
```
Comment (bdauria) [inline www/front_src/src/Login/index.test.tsx:125]:
```suggestion
  it(`submits the credentials when they are valid  and the "${labelConnect}" is clicked`, async () => {
```

File: www/front_src/src/Login/index.test.tsx [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+    );
+
+    await waitFor(() => {
+      expect(screen.getByText(labelLoginSucceeded)).toBeInTheDocument();
+    });
+
+    expect(window.location.href).toBe('http://localhost/monitoring/resources');
+  });
+
+  it(`submits invalid credentials when fields are filled and the "${labelConnect}" button is clicked`, async () => {
```
Comment (bdauria) [inline www/front_src/src/Login/index.test.tsx:150]:
```suggestion
  it(`does not submit the credentials when they are invalid and the "${labelConnect}" button is clicked`, async () => {
```

File: www/front_src/src/Login/index.test.tsx [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+    );
+
+    await waitFor(() => {
+      expect(screen.getByText(labelInvalidCredentials)).toBeInTheDocument();
+    });
+
+    expect(window.location.href).toBe('http://localhost/');
+  });
+
+  it('displays errors when fields are not emptied', async () => {
```
Comment (bdauria) [inline www/front_src/src/Login/index.test.tsx:182]:
```suggestion
  it('displays errors when fields are cleared', async () => {
```

File: www/front_src/src/Login/index.test.tsx [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+    );
+
+    await waitFor(() => {
+      expect(screen.getByLabelText(labelConnect)).toBeDisabled();
+    });
+
+    expect(screen.getAllByText(labelRequired)).toHaveLength(2);
+  });
+
+  it('displays the password when the corresponding icon button is clicked', () => {
```
Comment (bdauria) [inline www/front_src/src/Login/index.test.tsx:207]:
```suggestion
  it('displays the password when the corresponding action is clicked', () => {
```

File: www/front_src/src/Main/Provider.tsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+  children: React.ReactNode;
+}
+
+const Provider = ({ children }: Props): JSX.Element | null => {
+  const basename =
+    (document
+      .getElementsByTagName('base')[0]
+      ?.getAttribute('href') as string) || '';
+
+  const pathStartWithBasename = startsWith(basename, window.location.pathname);
```
Comment (bdauria) [inline www/front_src/src/Main/Provider.tsx:19]:
```suggestion
  const pathStartWsithBasename = startsWith(basename, window.location.pathname);
```

File: www/front_src/src/Main/Provider.tsx [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+  React.useEffect(() => {
+    if (pathStartWithBasename) {
```
Comment (bdauria) [inline www/front_src/src/Main/Provider.tsx:22]:
```suggestion
    if (pathStartsWithBasename) {
```

File: www/front_src/src/Main/Provider.tsx [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+      return;
+    }
+
+    const path = tail(window.location.pathname);
+    window.location.href = `${basename}${path}`;
+  }, []);
+
+  if (not(pathStartWithBasename)) {
```
Comment (bdauria) [inline www/front_src/src/Main/Provider.tsx:30]:
```suggestion
  if (not(pathStartsWithBasename)) {
```

File: www/front_src/src/Main/index.test.tsx [hunk 11]
```diff
@@ -0,0 +1,448 @@
...
+    );
+
+    await waitFor(() => {
+      expect(decodeURI(window.location.href)).toBe(
+        'http://localhost/install/upgrade.php',
+      );
+    });
+  });
+
+  it('does not the user to the upgrade page when the retrieved web versions contains an available version and the user is connected', async () => {
```
Comment (bdauria) [inline www/front_src/src/Main/index.test.tsx:345]:
```suggestion
  it('does not redirect the user to the upgrade page when the retrieved web versions contains an available version and the user is connected', async () => {
```

File: www/front_src/src/Main/Provider.tsx (hunk 8 above)
Comment (Thebarda) [reply www/front_src/src/Main/Provider.tsx:19]: Why did you add a "s" ?

**Review by bdauria — APPROVED** [review]

**Review by jdelpierre — APPROVED** [review]

**Review by Jordan91 — APPROVED** [review]

**Pull request merged**